test/utils.cpp

```diff
@@ -479,4 +479,32 @@ bool isSchedulerInUse(
   return false;
 }
 
+void validateSegmentation(
+    FusionKernelRuntime* runtime,
+    const std::vector<ScheduleHeuristic>& expected_heuristics) {
+  const auto& segment_groups = runtime->fusionSegments()->groups();
+
+  TORCH_CHECK(
+      segment_groups.size() == expected_heuristics.size(),
+      "Unexpected segments. Expected: ",
+      expected_heuristics.size(),
+      ". Actual: ",
+      segment_groups.size());
+
+  // Assumes up to two segments exist for simplicity
```

Comment:
Please note about this constraint. It's enough for the cases I intended.

Reply:
Comment added